### DISCUSSION TITLE
Datahub Importer - HTTP - Auth Injection from Env Var

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17755] [Q&A / Support]
When using the Datahub Importer to access an HTTP source, I am looking for a way to specify the [auth parameters (username and password)](https://pimcore.com/docs/platform/Data_Importer/Configuration/Data_Sources#http) from environmental variables. Two major reasons:
1. Avoid checking in secrets to git
2. Allow different configuration for staging and production

I know can just maintain local (untracked) copies of the Datahub importer, but I DO want to source control the settings themselves are there will be quite a few of them and I would prefer to minimize untracked files.

Is there anyway to run the the URL string through an environmental substitution and / or load that setting from a Pimcore / Symfony config?

**Comment by fashxp** [comment]
doesn't that help? https://pimcore.com/docs/platform/Datahub/Deployment#configuration-storage

**Reply by cancan101** [reply]
How would I use that to solve this problem? I am looking for the Datahub settings to be stored (read only) in a config file (as opposed to DB). However, I do want env var substitutions to happen when loading the file. Or at the very least the ability to load a subset of the settings values from the local environment (eg env var or symfony / pimcore config). I don't see that shown [here](https://pimcore.com/docs/platform/Datahub/Deployment/#sample-configuration-file).

**Reply by fashxp** [reply]
This shows exactly how you can configure Datahub to store/load configs as yaml files. 
And being standard symfony config files, it also should be possible to use standard mechanisms for environment variables...
